colossalai/nn/layer/moe/utils.py

```diff
@@ -51,6 +52,12 @@ def __call__(self, inputs: torch.Tensor):
         return inputs * noisy
 
 
+def autocast_softmax(logit: torch.Tensor, dim: int):
+    if logit.dtype != torch.float32:
+        logit = logit.float()
+    return F.softmax(logit, dim=dim)

```

Comment:
return F.softmax(logit.float() if logit.dtype != torch.float32 else logit, dim=dim)

Reply:
What's the necessity to write like that?

Reply:
That's OK